src/TransitionGroup.js

```diff
@@ -37,12 +37,31 @@ const propTypes = {
    * on individual children Transitions.
    */
   exit: PropTypes.bool,
+
+  staggerTime: PropTypes.number,
 };
 
 const defaultProps = {
   component: 'div',
 };
 
+function childMapperFn(child, index) {
+  const { staggerTime } = this.props;
+  const onExited = () => this.handleExited(child.key);
+  const style = staggerTime > 0 ? {
+    transitionDelay: `${staggerTime * (index + 1)}ms`
```

Comment:
This shouldn't go here, since it's css specific and this component is intended to be a generic component. I believe you should actually be able to implement this with the API as it is with minimal fuss. Your Transition children can accept a 'delay' property, and either compute it your as you map over the children or wrap TransitionGroup to inject it with similar logic.

```
<TransitionGroup>
  {items.map((item, idx) => 
    <StaggeredFade delay={staggerItem * (idx + 1)}>{item}</StaggeredFade>
   )}
</TransitionGroup>
```